packages/core/src/utils/logger.test.ts

```diff
@@ -0,0 +1,133 @@
+import { CreateLog } from '@logto/schemas';
+
+import { OmitAutoSetFields } from '@/database/utils';
+import { initTokenExchangeSuccessLogger } from '@/utils/logger';
+import { createContextWithRouteParameters } from '@/utils/test-utils';
+
+const insertLog: (data: OmitAutoSetFields<CreateLog>) => Promise<void> = jest.fn();
+
+jest.mock('@/queries/log', () => ({
+  insertLog: jest.fn(async (data) => insertLog(data)),
+}));
+
+describe('initTokenExchangeSuccessLogger', () => {
+  const ip = '192.168.0.1';
+  const userAgent =
+    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36';
+  const request = { ip, headers: { 'user-agent': userAgent } };
+
+  const userId = 'userId';
```

Comment:
nit: `userId` might be a little bit confusing. Suggest using something like `user_id_value` or `bar`